lib/netrc.c

```diff
@@ -81,8 +97,20 @@ static int parsenetrc(const char *host,
   file = fopen(netrcfile, FOPEN_READTEXT);
   if(file) {
     bool done = FALSE;
-    char netrcbuffer[4096];
-    int  netrcbuffsize = (int)sizeof(netrcbuffer);
+    char *netrcbuffer = NULL;
+    int  netrcbuffsize;
+
+    netrcbuffsize = get_file_sz(file);
+    if(netrcbuffsize++ == -1) {
```

Comment:
This will make having netrc as a (named) pipe fail.

Reply:
Fair point, as @dfandrich pointed out as well. This method is only applicable if `FILE *` is a seekable stream, which isn't necessarily always the case.